### PR TITLE
Use more gid ranges instead of one defined by min and max gid

### PR DESCRIPTION
**Pull request by stavamichal**
- new attribute for list of gid ranges used in all group* services
 - change behavior in gen scripts (preparing of file with ranges) and
   also in slave scripts (processing new files)

### CODE REVIEW

**Comment by stavamichal** [comment]
This pull request can be merged only after ht

**Comment by stavamichal** [comment]
This code can be merged only if new attributes with gid ranges will be set on all instances with GID namespaces and after new required attributes will be set everywhere for affected services (group, group_mu, group_nfs4).

File: gen/group [hunk 1]
```diff
@@ -60,17 +59,19 @@ foreach my $resourceData ($data->getChildElements) {
...
-close MIN_GID;
-close MAX_GID;
+my %gidRanges = %{$facilityAttributes{$A_FACILITY_GID_RANGES}};
+foreach my $minimum (sort { $a <=> $b } keys  %gidRanges) {
+	my $maximum = $gidRanges{$minimum};
+	for my $i ( $minimum .. $maximum ) {
+		print GID_RANGES $i, "\n";
+	}
+}
+close GID_RANGES;
```
Comment (licehammer) [inline gen/group]: You should check if file was closed correctly.

File: gen/group [hunk 2]
```diff
@@ -60,17 +59,19 @@ foreach my $resourceData ($data->getChildElements) {
...
-open MIN_GID,">$min_gid_file_name" or die "Cannot open $min_gid_file_name: $! \n";
-open MAX_GID,">$max_gid_file_name" or die "Cannot open $max_gid_file_name: $! \n";
+open GID_RANGES,">$gid_ranges_file_name" or die "Cannot open $gid_ranges_file_name\n";
 
-print MIN_GID $facilityAttributes{$A_FACILITY_MIN_GID}, "\n";
-print MAX_GID $facilityAttributes{$A_FACILITY_MAX_GID}, "\n";
-close MIN_GID;
-close MAX_GID;
+my %gidRanges = %{$facilityAttributes{$A_FACILITY_GID_RANGES}};
+foreach my $minimum (sort { $a <=> $b } keys  %gidRanges) {
```
Comment (licehammer) [inline gen/group]: The parameter for sort is not necessary.

**Pull request merged**